### ISSUE TITLE
sql: panic with window function in 19.1.5

### ISSUE THREAD

**Issue by mattcrdb** [cockroachdb/cockroach#42556]
**Describe the problem**

SQL panic in 19.1.5, client is restarted and node crashes.

Does not occur in 19.2.

**To Reproduce**

What did you do? Describe in your own words.
```sql
CREATE TABLE t1 (col1 INT8, col2 INT8, col3 INT8, col4 INT8, col5 INT8, col6 INT8, col7 INT8, col8 VARCHAR, col9 VARCHAR, col10 VARCHAR, col11 INT8, col12 INT8);

CREATE TABLE t2 (
	col1 INT8, col2 INT8, col3 INT8, col4 INT8, col5 INT8, col6 INT8, col7 INT8, col8 VARCHAR, col9 VARCHAR, col10 VARCHAR, col11 INT8, col12 INT8, col13 INT8, col14 INT8
);

INSERT
INTO
	t1
SELECT
	col1::INT8,
	col2::INT8,
	col3::INT8,
	col4::INT8,
	col5::INT8,
	col6::INT8,
	col7::INT8,
	col8::VARCHAR,
	col9::VARCHAR,
	col10::VARCHAR,
	col11::INT8,
	row_number() OVER (PARTITION BY col13 ORDER BY col14 DESC)::INT8
FROM
	t2;
```
**Expected behavior**
Insert to function properly.

**Environment:**
 - CockroachDB version 19.1.5
 - Client app: Cockroach SQL

**Additional context**
This is related to local mode.

Stack trace:
```
*
* ERROR: [n1,client=127.0.0.1:57582,user=root] a SQL panic has occurred while executing "INSERT INTO t1 SELECT col1::INT8, col2::INT8, col3::INT8, col4::INT8, col5::INT8, col6::INT8, col7::INT8, col8::VARCHAR, col9::VARCHAR, col10::VARCHAR, col11::INT8, row_number() OVER (PARTITION BY col13 ORDER BY col14 DESC)::INT8 FROM t2": windowNode can't be run in local mode
*
*
* ERROR: [n1,client=127.0.0.1:57582,user=root] a panic has occurred!
*
panic while executing 1 statements: INSERT INTO _ SELECT _::INT8, _::INT8, _::INT8, _::INT8, _::INT8, _::INT8, _::INT8, _::VARCHAR, _::VARCHAR, _::VARCHAR, _::INT8, row_number() OVER (PARTITION BY _ ORDER BY _ DESC)::INT8 FROM _

goroutine 1536 [running]:
runtime/debug.Stack(0x7289100, 0xc00ada7a40, 0xc000000003)
	/usr/local/go/src/runtime/debug/stack.go:24 +0xa7
github.com/cockroachdb/cockroach/pkg/util/log.ReportPanic(0x7289100, 0xc00ada7a40, 0xc000493300, 0x697c960, 0xc00d9bb140, 0x1)
	/go/src/github.com/cockroachdb/cockroach/pkg/util/log/crash_reporting.go:226 +0xb5
github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).closeWrapper(0xc000c38d80, 0x7289100, 0xc00ada7a40, 0x660d3a0, 0x7241b80)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/conn_executor.go:715 +0x2dd
github.com/cockroachdb/cockroach/pkg/sql.(*Server).ServeConn.func1(0xc000c38d80, 0x7289100, 0xc00ada7a40)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/conn_executor.go:434 +0x61
panic(0x660d3a0, 0x7241b80)
	/usr/local/go/src/runtime/panic.go:513 +0x1b9
github.com/cockroachdb/cockroach/pkg/sql.(*windowNode).Next(0xc009586c60, 0x72891c0, 0xc00dab9770, 0xc00b7c5000, 0xc000c390d0, 0xc00fd41988, 0x400cc5d, 0x6749d00)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/window.go:175 +0x39
github.com/cockroachdb/cockroach/pkg/sql.(*insertNode).BatchedNext(0xc00fd1c280, 0x72891c0, 0xc00dab9770, 0xc00b7c5000, 0xc000c390d0, 0x0, 0xab62440, 0x0)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/insert.go:471 +0xcc
github.com/cockroachdb/cockroach/pkg/sql.(*rowCountNode).startExec(0xc000426920, 0x72891c0, 0xc00dab9770, 0xc00b7c5000, 0xc000c390d0, 0x4016015, 0xc00ea65258)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/plan_batch.go:173 +0xd0
github.com/cockroachdb/cockroach/pkg/sql.startExec.func2(0x6b04308, 0x5, 0x728ae80, 0xc000426920, 0xc00fd41a98, 0x0)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/plan.go:496 +0x55
github.com/cockroachdb/cockroach/pkg/sql.(*planVisitor).visitInternal.func1(0xc00fab7ec0, 0x6b04308, 0x5, 0x728ae80, 0xc000426920)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/walk.go:146 +0x5d
panic(0x660d3a0, 0x7241b70)
	/usr/local/go/src/runtime/panic.go:513 +0x1b9
github.com/cockroachdb/cockroach/pkg/sql.(*windowNode).startExec(0xc009586c60, 0x72891c0, 0xc00dab9770, 0xc00b7c5000, 0xc000c390d0, 0xc00e019200, 0xc00fd41be8)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/window.go:171 +0x39
github.com/cockroachdb/cockroach/pkg/sql.startExec.func2(0x6b06f66, 0x6, 0x728b500, 0xc009586c60, 0xc0009d1260, 0xc00fd41bf8)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/plan.go:496 +0x55
github.com/cockroachdb/cockroach/pkg/sql.(*planVisitor).visitInternal.func1(0xc00fab7ec0, 0x6b06f66, 0x6, 0x728b500, 0xc009586c60)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/walk.go:146 +0x5d
github.com/cockroachdb/cockroach/pkg/sql.(*planVisitor).visitInternal(0xc00fab7ec0, 0x728b500, 0xc009586c60, 0x6b06f66, 0x6)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/walk.go:613 +0x1a2
github.com/cockroachdb/cockroach/pkg/sql.(*planVisitor).visit(0xc00fab7ec0, 0x728b500, 0xc009586c60, 0x6b063f0, 0x6)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/walk.go:113 +0x8d
github.com/cockroachdb/cockroach/pkg/sql.(*planVisitor).visitInternal(0xc00fab7ec0, 0x728aac0, 0xc00fd1c280, 0x6b063f0, 0x6)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/walk.go:452 +0x47c6
github.com/cockroachdb/cockroach/pkg/sql.(*planVisitor).visitConcrete(0xc00fab7ec0, 0x728aac0, 0xc00fd1c280)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/walk.go:126 +0x81
github.com/cockroachdb/cockroach/pkg/sql.(*planVisitor).visitInternal(0xc00fab7ec0, 0x728ae80, 0xc000426920, 0x6b04308, 0x5)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/walk.go:533 +0x4ceb
github.com/cockroachdb/cockroach/pkg/sql.(*planVisitor).visit(0xc00fab7ec0, 0x728ae80, 0xc000426920, 0x203003, 0x4010c39)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/walk.go:113 +0x8d
github.com/cockroachdb/cockroach/pkg/sql.walkPlan(0x72891c0, 0xc00dab9770, 0x728ae80, 0xc000426920, 0x0, 0x6c75f28, 0x0, 0x0, 0x0, 0xc00dab9860, ...)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/walk.go:77 +0x1a4
github.com/cockroachdb/cockroach/pkg/sql.startExec(0x72891c0, 0xc00dab9770, 0xc00b7c5000, 0xc000c390d0, 0x728ae80, 0xc000426920, 0xc00a046e58, 0xc00a046e50)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/plan.go:499 +0x116
github.com/cockroachdb/cockroach/pkg/sql.(*planNodeToRowSource).Start(0xc00fcfaa00, 0x72891c0, 0xc00dab9770, 0x90c9de0, 0x660c620)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/plan_node_to_row_source.go:124 +0xd9
github.com/cockroachdb/cockroach/pkg/sql/distsqlrun.(*ProcessorBase).Run(0xc00fcfaa00, 0x72891c0, 0xc00dab9770)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/distsqlrun/processors.go:800 +0x52
github.com/cockroachdb/cockroach/pkg/sql/distsqlrun.(*Flow).Run(0xc00fcf85a0, 0x72891c0, 0xc00dab9770, 0x6c75618, 0x0, 0x0)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/distsqlrun/flow.go:626 +0x1e9
github.com/cockroachdb/cockroach/pkg/sql.(*DistSQLPlanner).Run(0xc00081c780, 0xc00fab7ce0, 0xc009c7d710, 0xc00f2e0838, 0xc000ac58c0, 0xc000c39190, 0x0)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/distsql_running.go:252 +0x8a0
github.com/cockroachdb/cockroach/pkg/sql.(*DistSQLPlanner).PlanAndRun(0xc00081c780, 0x72891c0, 0xc00d97aae0, 0xc000c39190, 0xc00fab7ce0, 0xc009c7d710, 0x728ae80, 0xc000426920, 0xc000ac58c0)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/distsql_running.go:839 +0x227
github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).execWithDistSQLEngine(0xc000c38d80, 0x72891c0, 0xc00d97aae0, 0xc000c390d0, 0x2, 0xcb73dc0, 0xc0007d36b0, 0x0, 0x0, 0x0)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/conn_executor_exec.go:1125 +0x283
github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).dispatchToExecutionEngine(0xc000c38d80, 0x72891c0, 0xc00d97aae0, 0xc000c390d0, 0xcb73dc0, 0xc0007d36b0, 0x0, 0x0)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/conn_executor_exec.go:961 +0x658
github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).execStmtInOpenState(0xc000c38d80, 0x72891c0, 0xc00d97aae0, 0x728e800, 0xc00ed4b2c0, 0xc00dd0ea58, 0x1d2, 0x0, 0x0, 0x0, ...)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/conn_executor_exec.go:456 +0xdd0
github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).execStmt(0xc000c38d80, 0x72891c0, 0xc00d97aae0, 0x728e800, 0xc00ed4b2c0, 0xc00dd0ea58, 0x1d2, 0x0, 0x0, 0x0, ...)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/conn_executor_exec.go:102 +0x610
github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).run(0xc000c38d80, 0x7289100, 0xc00ada7a40, 0xc00004ee78, 0x5400, 0x15000, 0xc00004ef10, 0xc009bf9390, 0x0, 0x0)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/conn_executor.go:1186 +0x21d4
github.com/cockroachdb/cockroach/pkg/sql.(*Server).ServeConn(0xc0003018c0, 0x7289100, 0xc00ada7a40, 0xc000c38d80, 0x5400, 0x15000, 0xc00004ef10, 0xc009bf9390, 0x0, 0x0)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/conn_executor.go:436 +0xce
github.com/cockroachdb/cockroach/pkg/sql/pgwire.(*conn).processCommandsAsync.func1(0xc009fc88d9, 0xc00a0b9c00, 0x7289100, 0xc00ada7a40, 0xc009bf9390, 0xc0003018c0, 0xc00a9c1200, 0x728ce40, 0xc00a0b9be0, 0xc0010a8600, ...)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/pgwire/conn.go:580 +0x21f
created by github.com/cockroachdb/cockroach/pkg/sql/pgwire.(*conn).processCommandsAsync
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/pgwire/conn.go:515 +0x17b

*
* ERROR: [n1,client=127.0.0.1:57582,user=root] Reported as error 02ee48e10bef47448abcbcfe9833ab9b
*
driver: bad connection
warning: connection lost!
opening new connection: all session settings will be lost
warning: error retrieving the transaction status: dial tcp [::1]:26257: connect: connection refused
warning: connection lost!
opening new connection: all session settings will be lost
warning: error retrieving the database name: dial tcp [::1]:26257: connect: connection refused
```

**Comment by yuzefovich** [comment]
Same as #42247, but I will keep both issues open for now since the cluster version seem to be different.

**Comment by jordanlewis** [comment]
The problem here is something to do with the plan wrapping in distsql physical planner. For some reason, the `windowNode` isn't getting replaced by a distsql processor, which is an illegal state. I can't figure out why that would happen though. I would add some instrumentation around the `wrapPlan` code in the physical planner.

**Comment by yuzefovich** [comment]
Fixed by #42617.
